rfcs/0128-redeployable-clients.md

```diff
@@ -52,36 +55,123 @@ schemas:
-* refreshing taskcluster docs site, that displays information about taskcluster APIs
-* generating taskcluster command line tools, that provide command line interfaces to taskcluster APIs
+* building taskcluster clients, that provide language-level programming
+  interfaces to taskcluster APIs
+* refreshing taskcluster docs site, that displays information about taskcluster
+  APIs
+* generating taskcluster command line tools, that provide command line
+  interfaces to taskcluster APIs (such as `taskcluster-cli`)
 
 Note, other lifecycle stages then in turn depend on the above stages. For
-example, building workers depends on building a taskcluster client, as does
-building task authentication proxies, and other command line tools. Building
-taskcluster services depends on first building a taskcluster client to enable
-the service to communicate with other taskcluster services.
+example, the following activities depend on having a built taskcluster client:
+
+* building workers
+* building task authentication proxies
+* building command line tools such as those in the `generic-worker` repository
+* building taskcluster services that communicate with other taskcluster services
 
-The taskcluster platform delivery process currently respects the dependencies
-between lifecycle stages as follows:
+The development workflow typically looks like this:
 
 1. A service is implemented, which provides one or more APIs
-2. When the service is deployed, its API references are automatically
+2. When the new service is deployed, its API references are automatically
    published somewhere under https://references.taskcluster.net/ as part of the
    deployment process
-3. The [services manifest file](https://references.taskcluster.net/manifest.json) is manually updated to include the  API references
-
-....
-.... (WIP)
-....
+3. The [services manifest](https://references.taskcluster.net/manifest.json) is
+   manually updated to include the API references
+4. Taskcluster clients are built and released
+5. Software that interfaces with the new APIs is updated to pull in new client
+   versions, and released
+6. Released software is tested and deployed
 
 ### 3.1.2 Changes required
 
+1. If a service of a taskcluster deployment provides an HTTP interface, the
+   cluster should host the HTTP API reference document under
+   `<TASKCLUSTER_ROOT_URL>/references/<serviceName>/<version>/api.json`
+2. If a service of a taskcluster deployment provides an AMQP 0.9.1 interface,
+   the cluster should host the AMQP 0.9.1 API reference document under
+   `<TASKCLUSTER_ROOT_URL>/references/<serviceName>/<version>/exchanges.json`
+3. A taskcluster deployment should serve a service manifest
+   under `<TASKCLUSTER_ROOT_URL>/references/manifest.json` with the following
+   format:
+
+```
+{
+  "version" : 1,
+  "services": [
+    "<serviceName>": {
+      "api": true|false,
+      "exchanges": true|false,
+    },
```

Comment:
This seems to omit the API version.

Why switch from relative URLs to true/false?  Both are probably fine, but the URLs are a little more discoverable for people trying to navigate these files by hand.